### PR TITLE
feature: attesters and badges api

### PR DESCRIPTION
**Pull request by Kakktuss**

### CODE REVIEW

File: src/topics/attester/attester.ts [hunk 1]
```diff
@@ -1,20 +1,51 @@
 import { BigNumberish } from "ethers";
 import { AttestationsCollection } from "../attestations-collection/attestations-collection";
```
Comment (nicolas-geniteau) [inline src/topics/attester/attester.ts:2]: `import { AttestationsCollection } from "../attestations-collection/attestations-collection";`
->
`import { AttestationsCollection } from "../attestations-collection";`

File: src/topics/badge/badge.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { BigNumber, BigNumberish, ethers } from "ethers";
 import { Attester } from "../attester";
+import { ConstructedBadge } from "./types";
 
 export type BadgeConstructor = {
   name?: string;
```
Comment (nicolas-geniteau) [inline src/topics/badge/badge.ts:6]: name should not be mandatory ?

Below, in Badge constructor, `requirements` and `collectionId` are not used

**Review by nicolas-geniteau — CHANGES_REQUESTED** [review]

File: src/topics/badge/badge.ts [hunk 3]
```diff
@@ -28,6 +29,19 @@ export class Badge {
...
-    console.log("this.collectionId", this.collectionId);
+  }
+
+  public compute(claimId: BigNumberish, attester: Attester): ConstructedBadge {
+    this.computeCollectionId(claimId, attester);
+
+    return {
+      collectionId: this.collectionId ?? claimId,
+      metadata: {
+        name: this.name ?? "",
```
Comment (leosayous21) [inline src/topics/badge/badge.ts:40]: Name / description / Image should be mendatory

File: attesters/index.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import * as fs from "node:fs/promises";
```
[the remark this answers is not in the document]
Comment (leosayous21) [reply attesters/index.ts:1]: Yes, let's do it with a networks field for every network where it is deployed

File: attesters/hydra-s1-simple/index.ts [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
+import { AttesterNetwork } from "../../src/topics/attester/types";
+import { Badge } from "../../src/topics/badge";
 import { Group } from "../../src/topics/group";
 
-export const hydraS1SimpleAttester = async () =>
+export default async () =>
   new Attester({
-    collectionIdFirst: 10000001,
+    name: "hydra-s1-simple",
+    configurations: {
```
Comment (nicolas-geniteau) [inline attesters/hydra-s1-simple/index.ts:10]: rename `configurations` to `networkConfiguration` ?
Or have something like:
```
configuration: {
  network: {}
}
```
@leosayous21

File: attesters/hydra-s1-simple/index.ts [hunk 6]
```diff
@@ -1,11 +1,19 @@
 import { AttestationsCollection } from "../../src/topics/attestations-collection/attestations-collection";
```
Comment (nicolas-geniteau) [inline attesters/hydra-s1-simple/index.ts:1]: `import { AttestationsCollection } from "../../src/topics/attestations-collection";`

File: src/topics/attester/attester.test.ts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        AttesterNetwork.Rinkeby
+      );
+
+      expect(attesters).toMatchObject(expectedAttesters["rinkeby"]);
+
+      expect(attesters).toMatchObject(expectedAttesters["polygon"]);
+    });
+  });
+
+  describe("getConstructNetworkAttester", () => {
```
Comment (nicolas-geniteau) [inline src/topics/attester/attester.test.ts:140]: duplicated ??

File: src/topics/attester/attester.test.ts [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+import { Badge } from "../badge";
+import { Attester } from "./attester";
+import {
+  AllConstructedAttesters,
+  getConstructedAttesters,
+  getConstructedNetworkAttesters,
+} from "./attester.helper";
+import { AttesterNetwork } from "./types";
+
+jest.mock("../../../attesters/hydra-s1-simple", () => {
```
Comment (nicolas-geniteau) [inline src/topics/attester/attester.test.ts:14]: You should not use any mock here. I will show you an example with  group-generators (in another PR)

File: src/topics/badge/badge.api.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    )
+  );
+});
+
+router.get("/:network/:attester", async (req, res) => {
+  const { network, attester } = req.params;
+
+  res.json(
+    (
+      await getConstructNetworkAttester(network as AttesterNetwork, attester)
```
Comment (nicolas-geniteau) [inline src/topics/badge/badge.api.ts:27]: return 404 if not found ?

File: src/topics/badge/badge.api.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+const router = Router();
+
+router.get("/:network", async (req, res) => {
+  const { network } = req.params;
+
+  res.json(
+    Object.entries(
+      await getConstructedNetworkAttesters(network as AttesterNetwork)
+    ).map((attester) =>
+      attester[1].attestationsCollections.map((collection) => collection?.badge)
```
Comment (nicolas-geniteau) [inline src/topics/badge/badge.api.ts:17]: why `collection?.badge` and not `collection.badge` ?

**Review by nicolas-geniteau — CHANGES_REQUESTED** [review]

**Pull request closed without merging**